Duplicati/Server/Database/ServerSettings.cs

```diff
@@ -299,9 +303,41 @@ public void SetWebserverPassword(string password)
                     m_values[CONST.SERVER_PASSPHRASE] = pwd;
                     m_values[CONST.SERVER_PASSPHRASE_SALT] = salt;
                 }
+            }
 
-                SaveSettings();
+            SaveSettings();
+            GenerateWebserverPasswordTrayIcon();
+        }
+
+        public string WebserverPasswordTrayIcon => m_values[CONST.SERVER_PASSPHRASETRAYICON];
+
+        public string WebserverPasswordTrayIconHash => m_values[CONST.SERVER_PASSPHRASETRAYICONHASH];
+
+        public void GenerateWebserverPasswordTrayIcon()
+        {
+            var password = "";
+            var pwd = "";
+
+            if (!string.IsNullOrEmpty(m_values[CONST.SERVER_PASSPHRASE]))
+            {
+                password = Guid.NewGuid().ToString();
+                var buf = Convert.FromBase64String(m_values[CONST.SERVER_PASSPHRASE_SALT]);

```

Comment:
I don't think there is a guarantee that the `SERVER_PASSPHRASE_SALT` value is not empty?